addOns/graphql/src/main/java/org/zaproxy/addon/graphql/GraphQlParser.java

```diff
@@ -19,49 +19,108 @@
  */
 package org.zaproxy.addon.graphql;
 
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.reflect.TypeToken;
+import graphql.introspection.IntrospectionQuery;
+import graphql.introspection.IntrospectionResultToSchema;
+import graphql.language.Document;
+import graphql.schema.idl.SchemaPrinter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Map;
 import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.apache.commons.io.FileUtils;
 import org.apache.log4j.Logger;
-import org.parosproxy.paros.control.Control;
-import org.parosproxy.paros.extension.history.ExtensionHistory;
-import org.parosproxy.paros.model.HistoryReference;
-import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.Constant;
 import org.parosproxy.paros.network.HttpMessage;
-import org.parosproxy.paros.network.HttpSender;
-import org.zaproxy.zap.utils.ThreadUtils;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.zaproxy.zap.network.HttpRequestBody;
 
 public class GraphQlParser {
 
     private static final Logger LOG = Logger.getLogger(GraphQlParser.class);
+    private final URI endpointUrl;
+    private final int initiator;
```

Comment:
Can be removed not actually used.

Reply:
They have been used in the constructor and in the introspection method to send a request to the endpoint. They will also be used in the parse() method to send queries.

Reply:
I'm referring to the instance variable that's not being used, the parameter `initiator` is passed to the `RequestHandler`.